src/main/java/Parser.java

```diff
@@ -0,0 +1,87 @@
+import java.text.ParseException;
+
+public class Parser {
+
+    public Parser() {
+
+    }
+
+    /**
+     * Parses the sentence entered by the user and returns various commands
+     * @param input by the user
+     * @return other commands
+     */
+    // parse the command string keyed in by the user
+    public static Command parse (String input) throws DukeException {
+        String[] userInput = input.split(" ", 2);
+        String firstWord = userInput[0]; //firstword of the user input
+        //try {
+            switch (firstWord) {
+                case "bye":
+                    return new ExitCommand();
+                case "list":
+                    return new ListCommand();
+                case "done":
+                    int indexDone = Integer.parseInt(userInput[1]);
+                    return new DoneCommand(indexDone);
+                case "delete":
+                    int indexDelete = Integer.parseInt(userInput[1]);
+                    return new DeleteCommand(indexDelete);
+                case "massDelete":
+                    String[] indexRanges = userInput[1].split(" ", 2);
+                    int indexStart = Integer.parseInt(indexRanges[0]);
+                    int indexEnd = Integer.parseInt(indexRanges[1]);
+                    for (int i = 0; i < (indexEnd - indexStart); i++) {
+                        return new DeleteCommand(indexStart + i);
```

Comment:
This for loop will have at most 1 iteration, is this the intended behavior? 🥇

Reply:
whoops, I haven't implemented that. sorry!